server/src/main/java/com/saralein/server/Application.java

```diff
@@ -1,44 +1,49 @@
 package com.saralein.server;
 
-import com.saralein.server.controller.ErrorController;
-import com.saralein.server.controller.DirectoryController;
-import com.saralein.server.controller.FileController;
+import com.saralein.server.controller.Controller;
 import com.saralein.server.logger.Logger;
+import com.saralein.server.middleware.Middleware;
+import com.saralein.server.middleware.StaticMiddleware;
 import com.saralein.server.request.RequestParser;
 import com.saralein.server.response.ResponseSerializer;
 import com.saralein.server.router.Routes;
 import com.saralein.server.router.Router;
-
 import java.nio.file.Path;
 
 public class Application {
```

Comment:
I don't see any unit test for this. Did you have a reason for that? Was is hard, or maybe you didn't think this piece needed to be tested? Just curious